src/huggingface_hub/utils/endpoint_helpers.py

```diff
@@ -221,3 +236,10 @@ class ModelFilter:
     task: Optional[Union[str, List[str]]] = None
     trained_dataset: Optional[Union[str, List[str]]] = None
     tags: Optional[Union[str, List[str]]] = None
+
+    def __post_init__(self):
+        warnings.warn(
+            "ModelFilter is being deprecated and will be removed in huggingface_hub>=0.24. Please pass the filter parameters as keyword arguments directly to the `list_models` method.",
+            FutureWarning,
```

Comment:
Updated the category from `DeprecatedWarning` to `FutureWarning` because 
> DeprecatedWarning: Base category for warnings about deprecated features when those warnings are intended for other Python developers (ignored by default, unless triggered by code in __main__).
i.e. you won't get a warning in interepreter if you use the class as it is, unless we update the filters for warning.

Reply:
Yes that's perfect! This is also what we use in [`./utils/_deprecate.py`](https://github.com/huggingface/huggingface_hub/blob/133bf03ebf22e33f558a3a4041e4ec4cca6e9e00/src/huggingface_hub/utils/_deprecation.py) (no need to use these utils here). Nice idea to use the `__post_init__` method to deprecate dataclasses :)